monkey/monkey_island/cc/agent_event_handlers/node_update_facade.py

```diff
@@ -0,0 +1,90 @@
+from functools import lru_cache
+from ipaddress import IPv4Address, IPv4Interface
+
+from common.agent_events import AbstractAgentEvent
+from common.types import AgentID, MachineID
+from monkey_island.cc.models import CommunicationType, Machine
+from monkey_island.cc.repository import (
+    IAgentRepository,
+    IMachineRepository,
+    INodeRepository,
+    UnknownRecordError,
+)
+
+
+class NodeUpdateFacade:
```

Comment:
Should be renamed? This is as much about nodes as it is about machines. Maybe `NodeMachineUpdateFacade`

Reply:
How about `NetworkModelUpdateFacade`?